### PR TITLE
Export options added

### PR DESCRIPTION
**Pull request by Searay-330**
added back in export ability as well as added different options for profs to choose.
export everything - entire assignment dir.
export students files - handin directory not including the deleted submissions.
export autograde files - exports the autograder files.

right now perms for this are instructor or above we can change later if TA's need this.

### CODE REVIEW

File: app/controllers/assessments_controller.rb [hunk 1]
```diff
@@ -333,55 +333,44 @@ def getAssessmentVariable(key)
 
   # export - export an assessment by saving its persistent
   # properties in a yaml properties file.
+  action_auth_level :exportOptions, :instructor
+  def exportOptions
```
Comment (daviddob) [inline app/controllers/assessments_controller.rb]: Empty function definition?

File: app/controllers/assessments_controller.rb [hunk 2]
```diff
@@ -333,55 +333,44 @@ def getAssessmentVariable(key)
...
+
+  end
+
   action_auth_level :export, :instructor
   def export
-    base_path = Rails.root.join("courses", @course.name).to_s
-    asmt_dir = @assessment.name
-    begin
-      # Update the assessment config YAML file.
-      @assessment.dump_yaml
```
Comment (daviddob) [inline app/controllers/assessments_controller.rb]: Where did this end up in the final export?

File: app/controllers/assessments_controller.rb [hunk 3]
```diff
@@ -823,4 +812,114 @@ def valid_asmt_tar(tar_extract)
...
+      tarStream.close
+      send_data tarStream.string.force_encoding("binary"), filename: "#{@assessment.name}_#{Time.now.strftime('%Y%m%d')}.tar", content_type: "application/x-tar"
+    rescue SystemCallError => e
+      flash[:error] = "Unable to generate tarball -- #{e.message}"
+      redirect_to(:action => 'exportOptions') && return
+    rescue StandardError => e
+      flash[:error] = "Unable to generate tarball -- #{e.message}"
+      redirect_to(:action => 'exportOptions') && return 
+    else
+
```
Comment (daviddob) [inline app/controllers/assessments_controller.rb]: Empty else clause? Did you forget the success message?

File: app/controllers/assessments_controller.rb [hunk 4]
```diff
@@ -823,4 +812,114 @@ def valid_asmt_tar(tar_extract)
...
+    rescue SystemCallError => e
+      flash[:error] = "Unable to generate tarball -- #{e.message}"
+      redirect_to(:action => 'exportOptions') && return
+    rescue StandardError => e
+      flash[:error] = "Unable to generate tarball -- #{e.message}"
+      redirect_to(:action => 'exportOptions') && return 
+
+
+    else
+
```
Comment (daviddob) [inline app/controllers/assessments_controller.rb]: Another empty else clause?

File: app/controllers/assessments_controller.rb [hunk 5]
```diff
@@ -823,4 +812,114 @@ def valid_asmt_tar(tar_extract)
...
```
Comment (daviddob) [inline app/controllers/assessments_controller.rb]: "" See above

File: app/controllers/assessments_controller.rb (hunk 2 above)
Comment (Searay-330) [reply app/controllers/assessments_controller.rb]: this is part of the dump everything we dont use the yaml files so i didnt see a need to just export them all by them selfs

File: app/controllers/assessments_controller.rb (hunk 1 above)
Comment (Searay-330) [reply app/controllers/assessments_controller.rb]: refactored and its gone

File: app/controllers/assessments_controller.rb (hunk 3 above)
Comment (Searay-330) [reply app/controllers/assessments_controller.rb]: the "success message" is the download itself so ill delete the clause

File: app/controllers/assessments_controller.rb (hunk 4 above)
Comment (Searay-330) [reply app/controllers/assessments_controller.rb]: i copied and pasted from above

File: app/controllers/assessments_controller.rb (hunk 5 above)
Comment (Searay-330) [reply app/controllers/assessments_controller.rb]: i copied and pasted from above

**Pull request closed without merging**